app/javascript/controllers/search_convicts_controller.js

```diff
@@ -14,15 +13,30 @@ export default class extends ApplicationController {
     }
 
     search() {
+        if (this.query.length === 0) {
+            this.reset();
+            return;
+        }
+
         if (this.query == this.previousQuery) {
             return
         }
+
         this.previousQuery = this.query
 
+        const digitRegex = /\d/g;
+        const digitCount = (this.query.match(digitRegex) || []).length;
+        const nonDigitRegex = /(?!\+)\D/;
+        const containsNonDigit = nonDigitRegex.test(this.query);
+
+        if (digitCount < 4 && !containsNonDigit) {
+            return;
+        }
+
         this.abortPreviousFetchRequest()
 
         this.abortController = new AbortController()
-        fetch('/convicts/search?q=' + this.query, { signal: this.abortController.signal })
+        fetch('/convicts/search?q=' + encodeURIComponent(this.query), { signal: this.abortController.signal })
             .then(response => response.text())
             .then(html => {
                 this.handleResults(html)
```

Comment:
Pas le sujet de cette PR mais cette manière de changer le contenu est pas vraiment Rails Way. Ca fait le taff, on pourra en reparler pour une potentielle refonte dans le futur !

Reply:
C'était déjà fait "the Rails way" précédemment, avec le champ de recherche qui était un form_with il me semble, le formulaire submit avec Rails.fire(elem, 'submit') à la frappe, une partial pour les résultats etc... mais j'ai jamais su le faire fonctionner correctement. Le controller dans sa réponse wrappait toujours la partial avec des balises <html> superflues, pétant tout l'affichage. bref, je pense que ça sera plus simple avec Rails 7 et Turbo.